apps/angular/src/app/features/anime/anime-page/anime-page.component.html

```diff
@@ -1,66 +1,84 @@
-		class="mat-elevation-z8" 
-		aria-describedby="anime list"
-	>
-	
-		<ng-container matColumnDef="image">
-			<th mat-header-cell *matHeaderCellDef>Image</th>
-			<td mat-cell *matCellDef="let element"> 
-				<div class="anime-page__image-cell">
-					<img [src]="element.imageUrl" [alt]="element.titleEng + 'cover'" class="anime-page__image" /> 
-				</div>
-			</td>
-		</ng-container>
+	<form [formGroup]="filtersForm" class="anime-page__filters">
+		<mat-form-field>
+			<mat-label>Search</mat-label>
+			<input matInput placeholder="e.g. Naruto" [formControl]="filtersForm.controls.search">
+		</mat-form-field>
+		<mat-form-field>
+			<mat-label>Anime type</mat-label>
+			<mat-select [formControl]="filtersForm.controls.type" multiple>
+				<mat-option *ngFor="let type of animeTypes; trackBy: trackAnimeType" [value]="type">{{type}}</mat-option>
+			</mat-select>
+		</mat-form-field>
+	</form>
 
-		<ng-container matColumnDef="titleEnglish">
-			<th mat-header-cell *matHeaderCellDef>English title</th>
-			<td mat-cell *matCellDef="let element">{{element.titleEnglish}}</td>
-		</ng-container>
-		
-		<ng-container matColumnDef="titleJapanese">
-			<th mat-header-cell *matHeaderCellDef>Japanese title</th>
-			<td mat-cell *matCellDef="let element">{{element.titleJapanese}}</td>
-		</ng-container>
-		
-		<ng-container matColumnDef="aired.start">
-			<th mat-header-cell *matHeaderCellDef>Aired start</th>
-			<td mat-cell *matCellDef="let element">{{element.aired.start | date}}</td>
-		</ng-container>
+	<ng-container *ngIf="sorting$ | async as sorting">
+		<table *ngIf="animeList$ | async as animeList" 
+			mat-table 
+			matSort 
+			[matSortActive]="sorting.field"
+			[matSortDirection]="sorting.direction"
+			(matSortChange)="handleSortChange($event)"
+			[dataSource]="animeList.items" 
+			[trackBy]="trackById"
+			class="mat-elevation-z8" 
+			aria-describedby="anime list">
+			
+			<ng-container matColumnDef="image">
+				<th mat-header-cell *matHeaderCellDef>Image</th>
+				<td mat-cell *matCellDef="let element"> 
+					<div class="anime-page__image-cell">
+						<img [src]="element.imageUrl" [alt]="element.titleEng + 'cover'" class="anime-page__image" /> 
+					</div>
+				</td>
+			</ng-container>
+				
+			<ng-container matColumnDef="titleEnglish">
+				<th mat-header-cell *matHeaderCellDef mat-sort-header sortActionDescription>English title</th>
+				<td mat-cell *matCellDef="let element">{{element.titleEnglish}}</td>
+			</ng-container>
+				
+			<ng-container matColumnDef="titleJapanese">
+				<th mat-header-cell *matHeaderCellDef>Japanese title</th>
+				<td mat-cell *matCellDef="let element">{{element.titleJapanese}}</td>
+			</ng-container>
+				
+			<ng-container matColumnDef="aired.start">
+				<th mat-header-cell *matHeaderCellDef mat-sort-header sortActionDescription="Sort by aired start">Aired start</th>
+				<td mat-cell *matCellDef="let element">{{(element.aired.start !== null) ? (element.aired.start | date) : "" }}</td>
```

Comment:
I think it would be better, if we can avoid such hard to read expressions. I suggest to use something like that:

```suggestion
				<td mat-cell *matCellDef="let element">
				  <time 
				    *ngIf="element.aired.start as startDate; else emptyPlaceholder" 
				    [dateTime]="startDate
				  >
				    {{ startDate | date }}
				  </time>
				  
				  <ng-template #emptyPlaceholder>&mdash;</ng-template>
				</td>
```
Firstly, this approach allows you to make such logic clear to read. Also, you can move out `emptyPlaceholder` template to end of file and use it whenever, and it is good, because I believe it is good UX, when user clear see: there is `-` (no value) instead only empty space. Besides, in my example, we are keeping HTML semantic when we're using `time` for date properties.